compiler/include/symbol.h

```diff
@@ -556,6 +556,33 @@ class TypeSymbol : public Symbol {
 *                                                                             *
 ************************************** | *************************************/
 
+class InterfaceSymbol : public Symbol {
```

Comment:
let's put a comment describing the purpose of this AST class / what syntax it supports